### PR TITLE
Metadata: Bring back SqlServerDbFunctionConvention

### LINKED ISSUE #20182: Metadata: Bring back SqlServerDbFunctionConvention
To set schema as dbo conventionally.

It was there in 2.2 release. It got removed at some point. Currently query injects dbo when printing.

### PR DESCRIPTION
**Pull request by smitpatel**
Resolve #20182

### CODE REVIEW

File: src/EFCore.SqlServer/Metadata/Conventions/SqlServerConventionSetBuilder.cs [hunk 1]
```diff
@@ -93,6 +93,8 @@ public override ConventionSet CreateConventionSet()
             ReplaceConvention(conventionSet.ModelFinalizingConventions,
                 (SharedTableConvention)new SqlServerSharedTableConvention(Dependencies, RelationalDependencies));
```
Comment (AndriySvyryd) [inline src/EFCore.SqlServer/Metadata/Conventions/SqlServerConventionSetBuilder.cs:95]: Remove blank line

Comment (smitpatel) [reply src/EFCore.SqlServer/Metadata/Conventions/SqlServerConventionSetBuilder.cs:95]: 😢

File: src/EFCore.SqlServer/Metadata/Conventions/SqlServerDbFunctionConvention.cs [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+using JetBrains.Annotations;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+using Microsoft.EntityFrameworkCore.Metadata.Conventions.Infrastructure;
+using Microsoft.EntityFrameworkCore.Utilities;
+
+// ReSharper disable once CheckNamespace
+namespace Microsoft.EntityFrameworkCore.Metadata.Conventions
+{
+    /// <summary>
+    ///     A convention that ensures that <see cref="IDbFunction.Schema"/> is populated for database functions which are not built-in.
```
Comment (AndriySvyryd) [inline src/EFCore.SqlServer/Metadata/Conventions/SqlServerDbFunctionConvention.cs:13]: How do you determine that they are built-in?

Comment (smitpatel) [reply src/EFCore.SqlServer/Metadata/Conventions/SqlServerDbFunctionConvention.cs:13]: That is coming in https://github.com/dotnet/efcore/issues/17268 (submitting PR for that once this is approved).

Just future-proof documentation. 😄

Comment (smitpatel) [reply src/EFCore.SqlServer/Metadata/Conventions/SqlServerDbFunctionConvention.cs:13]: I can improve this in next PR by referring the IsBuiltIn property too.

**Pull request merged**